internal/config/common.go

```diff
@@ -50,35 +53,7 @@ func GetConfiguration[T any](filename string, defaultGenerator func() T) (T, err
 	var config T
 
 	if _, err := toml.DecodeFile(configPath, &config); err != nil {
-		return defaultGenerator(), nil
-	}
-	return config, nil
-}
-
-func CreateTempConfigFile(content string, t *testing.T) (string, func()) {
-	t.Helper()
-
-	exePath, err := os.Executable()
-
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	tmpfile, err := os.CreateTemp(filepath.Dir(exePath), "config.toml")
-
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if _, err := tmpfile.Write([]byte(content)); err != nil {
-		t.Fatal(err)
-	}
-
-	if err := tmpfile.Close(); err != nil {
-		t.Fatal(err)
-	}
-
-	return tmpfile.Name(), func() {
-		os.Remove(tmpfile.Name())
+		return defaultGenerator(), err
```

Comment:
This would rather be the zero value than the default, as it shouldn't be accessed.

Reply:
Given it shouldn't be accessed I'm not sure it matters too much

Reply:
yeah, probably should have annotated with `NIT`. The advantage is that if you do ignore the error, you'll end up with a nonsense config, so the bug is more likely to be surfaced